test/models/ott/CategorisationInfoSpec.scala

```diff
@@ -536,10 +538,277 @@ class CategorisationInfoSpec extends SpecBase {
+
+          "when there are Category 1 that need answers but there are Category 2 assessment with no exemptions" in {
+            val mockOttResponse = OttResponse(
+              GoodsNomenclatureResponse(
+                "some id",
+                "1234567890",
+                None,
+                Instant.EPOCH,
+                None,
+                List("test")
+              ),
+              categoryAssessmentRelationships = Seq(
+                CategoryAssessmentRelationship("assessmentId2"),
+                CategoryAssessmentRelationship("assessmentId1")
+              ),
+              includedElements = Seq(
+                ThemeResponse("themeId1", 1),
+                CategoryAssessmentResponse(
+                  "assessmentId2",
+                  "themeId1",
+                  Seq(
+                    ExemptionResponse("exemptionId1", ExemptionType.Certificate),
+                    ExemptionResponse("exemptionId2", ExemptionType.AdditionalCode)
+                  )
+                ),
+                ThemeResponse("themeId2", 2),
+                CertificateResponse("exemptionId1", "code1", "description1"),
+                AdditionalCodeResponse("exemptionId2", "code2", "description2"),
+                ThemeResponse("ignoredTheme", 3),
+                CertificateResponse("ignoredExemption", "code3", "description3"),
+                CategoryAssessmentResponse(
+                  "assessmentId1",
+                  "themeId2",
+                  Seq.empty
+                )
+              ),
+              descendents = Seq.empty[Descendant]
+            )
+
+            val expectedAssessmentId1 = CategoryAssessment(
+              "assessmentId1",
+              2,
+              Seq.empty
+            )
+            val expectedAssesmentId2  = CategoryAssessment(
+              "assessmentId2",
+              1,
+              Seq(
+                Certificate("exemptionId1", "code1", "description1"),
+                AdditionalCode("exemptionId2", "code2", "description2")
+              )
+            )
+
+            val expectedAssessments = Seq(
+              expectedAssesmentId2,
+              expectedAssessmentId1
+            )
+
+            val expectedAssessmentsThatNeedAnswers = Seq(expectedAssesmentId2)
+
+            val expectedResult =
+              CategorisationInfo(
+                "1234567890",
+                expectedAssessments,
+                expectedAssessmentsThatNeedAnswers,
+                None,
+                0,
+                isNiphlAuthorised = true
+              )
+
+            val result = CategorisationInfo.build(mockOttResponse, "1234567890", testTraderProfileResponseWithNiphl)
+            result.value mustEqual expectedResult
+          }
+
+          "when there are Category 1 that needed answer and there is a NIPHL assessment but there is a Category 2 assessment with no exemptions" in {
```

Comment:
pedantic: but should be and in the title

Reply:
Updated